File: pm_core/src/infrastructure/store/mapper/label_mapper.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type LabelMapper struct{}

func NewLabelMapper() *LabelMapper {
	return &LabelMapper{}
}

func (m *LabelMapper) ToEntity(mdl *model.LabelModel) *entity.LabelEntity {
	if mdl == nil {
		return nil
	}

	return &entity.LabelEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
		},
		ProjectID:    mdl.ProjectID,
		Name:         mdl.Name,
		Color:        mdl.Color,
		ActiveStatus: mdl.ActiveStatus,
	}
}

func (m *LabelMapper) ToModel(e *entity.LabelEntity) *model.LabelModel {
	if e == nil {
		return nil
	}

	return &model.LabelModel{
		BaseModel: model.BaseModel{
			ID: e.ID,
		},
		ProjectID:    e.ProjectID,
		Name:         e.Name,
		Color:        e.Color,
		ActiveStatus: e.ActiveStatus,
	}
}

func (m *LabelMapper) ToEntities(models []*model.LabelModel) []*entity.LabelEntity {
	if models == nil {
		return nil
	}

	entities := make([]*entity.LabelEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}

	return entities
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/LabelMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.LabelEntity;
import serp.project.pmcore.infrastructure.store.model.LabelModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class LabelMapper extends BaseMapper {

    public LabelEntity toEntity(LabelModel model) {
        if (model == null) { return null; }
        return LabelEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .projectId(model.getProjectId())
                .name(model.getName())
                .color(model.getColor())
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public LabelModel toModel(LabelEntity entity) {
        if (entity == null) { return null; }
        return LabelModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .projectId(entity.getProjectId())
                .name(entity.getName())
                .color(entity.getColor())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<LabelEntity> toEntities(List<LabelModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }
}
